src/components/scenes/RampSelectOptionScene.tsx

```diff
@@ -0,0 +1,389 @@
+  const { rampQuoteRequest } = route.params
+
+  const theme = useTheme()
+  const account = useSelector(state => state.core.account)
+  // Get ramp plugins
+  const { data: rampPluginArray = [], isLoading: isPluginsLoading } =
+    useRampPlugins({ account })
+  const rampPlugins = React.useMemo(() => {
+    const map: Record<string, RampPlugin> = {}
+    for (const plugin of rampPluginArray) {
+      map[plugin.pluginId] = plugin
+    }
+    return map
+  }, [rampPluginArray])
+
+  // Use supported plugins hook
+  const { supportedPlugins } = useSupportedPlugins({
+    selectedWallet: rampQuoteRequest.wallet,
+    selectedCrypto:
+      rampQuoteRequest.wallet != null
+        ? {
+            pluginId: rampQuoteRequest.pluginId,
+            tokenId: rampQuoteRequest.tokenId
+          }
+        : undefined,
+    selectedFiatCurrencyCode: rampQuoteRequest.fiatCurrencyCode.replace(
+      'iso:',
+      ''
+    ),
+    countryCode: rampQuoteRequest.regionCode?.countryCode,
+    stateProvinceCode: rampQuoteRequest.regionCode?.stateProvinceCode,
+    plugins: rampPlugins,
+    direction: rampQuoteRequest.direction
+  })
+
+  // Use supported plugins
+  const pluginsToUse = Object.fromEntries(
+    supportedPlugins.map(result => [result.plugin.pluginId, result.plugin])
+  )
+
+  // Use the ramp quotes hook
+  const {
+    quotes: allQuotes,
+    isLoading: isLoadingQuotes,
+    isFetching: isFetchingQuotes,
+    errors: failedQuotes
+  } = useRampQuotes({
+    rampQuoteRequest,
+    plugins: pluginsToUse
+  })
+
+  const handleQuotePress = async (quote: RampQuoteResult): Promise<void> => {
+    try {
+      await quote.approveQuote({
+        coreWallet: rampQuoteRequest.wallet!
+      })
+    } catch (error) {
+      console.error('Failed to approve quote:', error)
+    }
+  }
+
+  // Get the best quote overall
+  const bestQuoteOverall = allQuotes[0]
+
+  // Group quotes by payment type and sort within each group
+  const quotesByPaymentType = React.useMemo(() => {
+    const grouped = new Map<string, RampQuoteResult[]>()
+
+    allQuotes.forEach(quote => {
+      const paymentType = quote.paymentType
+      const existing = grouped.get(paymentType) ?? []
+      grouped.set(paymentType, [...existing, quote])
+    })
+
+    // Sort quotes within each payment type group
+    grouped.forEach(quotes => {
+      quotes.sort((a, b) => {
+        const cryptoAmountA = parseFloat(a.cryptoAmount)
+        const cryptoAmountB = parseFloat(b.cryptoAmount)
+
+        // Guard against division by zero
+        if (cryptoAmountA === 0 || cryptoAmountB === 0) {
+          // If either crypto amount is zero, sort that quote to the end
+          if (cryptoAmountA === 0 && cryptoAmountB === 0) return 0
+          if (cryptoAmountA === 0) return 1
+          return -1
+        }
+
+        const rateA = parseFloat(a.fiatAmount) / cryptoAmountA
+        const rateB = parseFloat(b.fiatAmount) / cryptoAmountB
+        return rateA - rateB
+      })
```

Comment:
Within each payment-type group, quotes are sorted by ascending fiat/crypto rate for both directions. For sells, the best rate is the highest fiat-per-crypto, so the sort should reverse. This affects which provider is selected by default in each card.

Reply:
Good catch, though it's actually the best fiat-per-crypto for _buys and sells_ only if the source amount is denominated in crypto (because we want to find the highest fiat for our fixed crypto amount). This is an interesting miss on my testing, because I've been testing with fiat amounts mostly which is why I never noticed the sorting issue when testing buy/sell quoting! Thanks for finding this one.